### DISCUSSION TITLE
Some Actions formatting Question.

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#137339] [Actions] [answered]
### Select Topic Area

Question

### Body

I have some small action formattin (yaml formatting question here).

I currently have this if-statement on the job-level:
```yaml
if: ${{ !github.event.issue.pull_request && contains(github.event.comment.body, 'Duplicate of') && (github.event.comment.author_association == 'OWNER' || github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'COLLABORATOR') }}
```
As you can see is it quite long and a bit hard to read.
So I am now asking, if it is fine to split the if-statement up into a multi-line YAML text, using the `|` character like this:
```yaml
if: |
  !github.event.issue.pull_request &&
  contains(github.event.comment.body, 'Duplicate of') &&
  (github.event.comment.author_association == 'OWNER' || github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'COLLABORATOR')
```
I assume it is safe to do so, as the final string would afaik be a single string with spaces per new line in the end, but I want to ask anyways to be safe here.

**Comment by ChristopherHX** [comment, marked as answer]
Based on my experience, yes this is safe for conditions that don't contain any `${{`.

If you would do this for non string fields with surrounding `${{ ... }}` make shure to use `|-` that removes the final newline, which avoids an implicit string cast.

ok as well (invalid without the `-` after the `|`)
```yaml
if: |-
  ${{
    !github.event.issue.pull_request &&
    contains(github.event.comment.body, 'Duplicate of') &&
    (github.event.comment.author_association == 'OWNER' || github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'COLLABORATOR')
  }}
```

E.g. Only if condition can omit the expr guards
```yaml
continue-on-error: |-
  ${{
    !github.event.issue.pull_request &&
    contains(github.event.comment.body, 'Duplicate of') &&
    (github.event.comment.author_association == 'OWNER' || github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'COLLABORATOR')
  }}

```

String literals can contain newlines and are preserved as `\n`.

It will get exciting starting with December, GitHub plans to replace the actions dotnet service by a new one rewritten in golang. SRC https://github.com/actions/runner/issues/1182#issuecomment-2317953582

> spaces per new line

Hmm that syntax preserves the newline char, only the \r is dropped.

The other fold syntax (`>`) used to be broken in GitHub Actions, which usually replaces newlines by space and is currently the same as `|` if I'm not out of date.

**Reply by Andre601** [reply]
Thanks for the detailed reply. Helped me a lot. ❤️
